Play sound on new request

phikal opened the pull request:
Im IRC wurde erwähnt das so was nützlich sein könnte, aber ich konnte es nicht selbst testen.

phikal committed:
Play sound on new request

noctux commented:
Hey,

thank you for your pullrequest! Nice idea. However, I do not really like the idea of playing sound unconditionally. So we would probably need a way to store preferences and have sound off by default? The proper way for that should be browser's localstorage and load the setting 'display sound' from there (I do not really want to have user-changeable settings serverside, as that would suddenly entail using a database...).

I've committed a basic framework for storing preferences in e22fff9. That way, you could add a Button to the [button area in the navbar](https://github.com/noctux/adora-belle/blob/e22fff98863ff42e70de0f71dcce2cced6289574/www/index.html#L19) like [this](https://github.com/noctux/adora-belle/blob/e22fff98863ff42e70de0f71dcce2cced6289574/www/index.html#L198), or (if you implement that feature for the user side as well simply as static html? That way, we could notify users when they were moved to the "serviceRequests" section).

On another note: what is the license of the statically included opus file? Is it in the public domain?

Best regards,
Simon

noctux commented:
Oh, and it would probably suffice to ping once per updaterun, even if there were multiple new requests? Otherwise, the initial load of a busy session might easily get onto one's nerves?

phikal commented:
I'll try to set what I can do. 

> On another note: what is the license of the statically included opus file? Is it in the public domain?

I found it on some Stackoverflow answer (was wav, but I converted and optimized it), so I don't really know. It should be trivial enough to re-create it though. 

And on that topic, would you mind if I added the opus sound directly to the repo, instead of embedding it into the html file?

noctux commented:
> I'll try to set what I can do.

Cool, ping me if you need any help!

> > On another note: what is the license of the statically included opus file? Is it in the public domain?
> 
> I found it on some Stackoverflow answer (was wav, but I converted and optimized it), so I don't really know. It should be trivial enough to re-create it though.

Yeah, we probably should go down that road or look for another media-file with a known, compatible (preferably public domain) license.

> And on that topic, would you mind if I added the opus sound directly to the repo, instead of embedding it into the html file?

No, please do that. `./www/static` should be the right place, I think.

noctux commented:
Just a small update: As we now have localStorage, I've implemented storing login credentials. The configuration is done using a new "modal" (the fancy bootstrap term for dialog boxes) that is triggered using a preference menu button. So now, adding the sound preference should be as easy as adding a new, suitable entry [into the config object here](https://github.com/noctux/adora-belle/commit/84f772611c06ad5246a707c0b1c7f34ab26ead01#diff-ab469c2203f00940178ab4d78eacaad6R194). Leaving the handler undefined will use the default handler, which simple calls `setPreference` with the respective checkbox value. The name of the entry should match the value in the settings object.

If you've already spent time on that button-stuff and do not want to do the conversion to modals, just  include the button in the pullrequest and I will do the conversion for you :)

phikal commented:
Don't worry, the only thing I managed to get to was creating a sound file. I'm going to try to implement what you described today in a new pull request.
